ray_write_shapefiles.py

```diff
@@ -0,0 +1,169 @@
+import os
+from datetime import datetime
+import shutil
+from io import BytesIO 
+from typing import Any, Dict
+
+import shapefile
+from osgeo import gdal, osr
+from shapely.geometry import Polygon, Point
+
+import gdal_virtual_file_system as gdal_vfs
+from mpl_config import MPL_Config
+
+
+def delete_and_create_dir(dir: str):
+    try:
+        shutil.rmtree(dir)
+    except Exception as e:
+        print(f"Error deleting directory {dir}: {e}")
+
+    try:
+        os.makedirs(dir, exist_ok=True)
+        print(f"Created directory {dir}")
+    except Exception as e:
+        print(f"Error creating directory {dir}: {e}")
+
+
+class WriteShapefiles:
+    def __init__(
+        self,
+        config: MPL_Config
+    ):
+        self.config = config
+        self.current_timestamp_str = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+        if self.config.GCP_FILESYSTEM is None:
+            delete_and_create_dir(self.config.RAY_SHAPEFILES)
+
+    def get_coordinate_system_info(self, file_path: str, file_bytes: bytes):
```

Comment:
seems to be a private function, see my comment below!

Reply:
done